Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+
+kubectl label node node-2-2-1 topology.rook.io/room=room-2
+kubectl label node node-2-2-1 topology.rook.io/rack=rack-2-2
+kubectl label node node-2-2-1 topology.rook.io/chassis=node-2-2-1
+
+kubectl label node node-2-2-2 topology.rook.io/room=room-2
+kubectl label node node-2-2-2 topology.rook.io/rack=rack-2-2
+kubectl label node node-2-2-2 topology.rook.io/chassis=node-2-2-2
+
+# -- room-f (failure domain) --
+
+kubectl label node node-f-1-1 topology.rook.io/room=room-f
+kubectl label node node-f-1-1 topology.rook.io/rack=rack-f-1
+kubectl label node node-f-1-1 topology.rook.io/chassis=node-f-1-1
+```
+
+## Planning the Nodes Where Ceph Daemons Will Run
+
+### Ceph MONs
+Using the hypothetical data center described in the [0verview](#overview), this section will look at
+planning the nodes where Ceph daemons are going to run.
+
+Ceph MON scheduling is one of the more detailed, and more important, things to understand about
+maintaining a healthy Ceph cluster. The goals we will target in this section can be summarized as:
+“Avoid risky co-location scenarios, but allow them if there are no other options in order to still
+have as much redundancy as possible.”
+
+This can lead us to the following specific goals:
+ * Allow MONs to be in the same room if a room is unavailable.
+ * Allow MONs to be in the same rack if no other racks in the room are available.
+ * Allow MONs to be on the same host only if no other hosts are available.</br>
+   We must allow host co-location specifically in the cluster configuration `cluster.yaml` by
+   setting `allowMultiplePerNode: true`.
+   > **IMPORTANT:** This **cannot** be set to true for clusters using host networking.
+
+> **TIP:**  We recommend using the same topology labels used for informing the CRUSH map here for
+> convenience.
+
+Because of the MON SSD availability in our hypothetical data center, we only want monitors to be
+able to run where shown below in green. We need to plan for monitors to fail over, and so we will
+make two nodes explicitly available for this scenario. In our example, we want any node with a MON
+SSD to be a MON failover location in emergencies, for maximum cluster health. This is highlighted in
+orange below. This will give us the most redundancy under failure conditions.
+
+![Example Datacenter - Mon Failover](media/ceph-bp-example-datacenter-mon-failover.png)
+
+To implement this in Rook, we ensure that Rook will only schedule MONs on nodes with MON SSDs. We
+therefore have a required Affinity for those nodes. We can accomplish this by applying a
+`ceph-mon-ssd=true` label to nodes with SSDs for Ceph MONs. Note that the `mon` section’s
+`nodeAffinity` takes precedence over and overwrites the `all` section’s `nodeAffinity` for Ceph MON
+daemons. Make sure that you re-specify the rules from the `all` section to ensure Ceph MONs also
+continue to maintain Affinity only for storage nodes as we configured above.
+
+
+**`cluster.yaml`:**
```

Comment:
This whole section on mon affinity and pod anti-affinity feels like it belongs in ceph-cluster-crd.md. We already have some similar examples to this. It would be good to enhance those examples if needed for this scenario. Then this doc can stay focused on the implementation of this advanced example.